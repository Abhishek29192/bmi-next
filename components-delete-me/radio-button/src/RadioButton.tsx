import React from "react";
import ButtonBase from "@material-ui/core/ButtonBase";
import classnames from "classnames";
import styles from "./RadioButton.module.scss";

export type Props = {
  name?: string;
  value: string;
  id?: string;
  children: React.ReactNode;
  before?: React.ReactNode;
  className?: string;
} & React.InputHTMLAttributes<HTMLInputElement>;

const RadioButton = ({
  name,
  value,
  id = `radio-${name}-${value}`,
  children,
  before,
  className,
  style,
  disabled,
  ...rest
}: Props) => {
  return (
    <div
      className={classnames(
        styles["RadioButton"],
        { [styles["RadioButton--disabled"] as string]: disabled },
        className
      )}
      style={style}
    >
      <input
        id={id}
        className={styles["input"]}
        type="radio"
        name={name}
        value={value}
        disabled={disabled}
        {...rest}
      />
      <label className={styles["label"]} htmlFor={id}>
        {before}
        <ButtonBase
          tabIndex={-1}
          component="div"
          className={styles["text"]}
          disabled={disabled}
        >
          {children}
        </ButtonBase>
      </label>
    </div>
  );
};

export default RadioButton;
