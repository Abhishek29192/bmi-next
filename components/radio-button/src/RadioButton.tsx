import React from "react";
import ButtonBase from "@material-ui/core/ButtonBase";
import classnames from "classnames";
import styles from "./RadioButton.module.scss";

export type Props = {
  name?: string;
  value: string;
  id?: string;
  children: React.ReactNode;
  before?: React.ReactNode;
  className?: string;
} & React.InputHTMLAttributes<HTMLInputElement>;

const RadioButton = ({
  name,
  value,
  id = `radio-${name}-${value}`,
  children,
  before,
  className,
  style,
  ...rest
}: Props) => {
  return (
    <div className={classnames(styles["RadioButton"], className)} style={style}>
      <input
        id={id}
        className={styles["input"]}
        type="radio"
        name={name}
        value={value}
        {...rest}
      />
      <label className={styles["label"]} htmlFor={id}>
        {before}
        <ButtonBase tabIndex={-1} component="div" className={styles["text"]}>
          {children}
        </ButtonBase>
      </label>
    </div>
  );
};

export default RadioButton;
