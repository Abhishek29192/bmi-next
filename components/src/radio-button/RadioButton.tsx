import { ButtonBase } from "@material-ui/core";
import classnames from "classnames";
import React from "react";
import { useStyles } from "./styles";

export type Props = {
  name?: string;
  value: string;
  id?: string;
  children: React.ReactNode;
  before?: React.ReactNode;
  className?: string;
} & React.InputHTMLAttributes<HTMLInputElement>;

const RadioButton = ({
  name,
  value,
  id = `radio-${name}-${value}`,
  children,
  before,
  className,
  style,
  disabled,
  ...rest
}: Props) => {
  const classes = useStyles();
  return (
    <div
      className={classnames(
        classes.root,
        disabled && classes.disabled,
        className
      )}
      style={style}
    >
      <input
        id={id}
        className={classes.input}
        type="radio"
        name={name}
        value={value}
        disabled={disabled}
        {...rest}
      />
      <label className={classes.label} htmlFor={id}>
        {before}
        <ButtonBase
          tabIndex={-1}
          component="div"
          className={classes.text}
          disabled={disabled}
        >
          {children}
        </ButtonBase>
      </label>
    </div>
  );
};

export default RadioButton;
