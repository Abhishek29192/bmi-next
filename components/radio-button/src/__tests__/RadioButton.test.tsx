import React from "react";
import { render } from "@testing-library/react";
import RadioButton from "../";

describe("RadioButton component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <RadioButton name="deckType" value="Combustible (timber/plywood)">
        Combustible (timber/plywood)
      </RadioButton>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("render with prefix", () => {
    const { container } = render(
      <RadioButton
        name="color"
        before={
          <div
            style={{
              minHeight: "100%",
              minWidth: 72,
              backgroundColor: "#D9D8D8"
            }}
          />
        }
        value="Grey"
      >
        Grey
      </RadioButton>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("calls onChange", () => {
    const handleOnChange = jest.fn();
    const text = "Combustible (timber/plywood)";

    const { container } = render(
      <RadioButton
        onChange={handleOnChange}
        name="deckType"
        value="Combustible (timber/plywood)"
      >
        {text}
      </RadioButton>
    );

    container.querySelector("label").click();

    expect(handleOnChange).toHaveBeenCalled();
  });
});
