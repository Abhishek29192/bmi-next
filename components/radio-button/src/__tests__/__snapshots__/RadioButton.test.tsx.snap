// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton component render with prefix 1`] = `
<div>
  <div
    class="RadioButton"
  >
    <input
      class="input"
      id="radio-color-Grey"
      name="color"
      type="radio"
      value="Grey"
    />
    <label
      class="label"
      for="radio-color-Grey"
    >
      <div
        style="min-height: 100%; min-width: 72px; background-color: rgb(217, 216, 216);"
      />
      <div
        aria-disabled="false"
        class="MuiButtonBase-root text"
        role="button"
        tabindex="-1"
      >
        Grey
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </label>
  </div>
</div>
`;

exports[`RadioButton component renders correctly 1`] = `
<div>
  <div
    class="RadioButton"
  >
    <input
      class="input"
      id="radio-deckType-Combustible (timber/plywood)"
      name="deckType"
      type="radio"
      value="Combustible (timber/plywood)"
    />
    <label
      class="label"
      for="radio-deckType-Combustible (timber/plywood)"
    >
      <div
        aria-disabled="false"
        class="MuiButtonBase-root text"
        role="button"
        tabindex="-1"
      >
        Combustible (timber/plywood)
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </label>
  </div>
</div>
`;
